frontend/src/views/Results.tsx

```diff
@@ -1,10 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useLocation, useHistory } from 'react-router-dom';
+import { Message } from 'stompjs';
+import { LargeText, Text } from '../components/core/Text';
+import { Game, Player, playAgain } from '../api/Game';
+import { checkLocationState } from '../util/Utility';
+import { errorHandler } from '../api/Error';
+import PlayerResultsCard from '../components/card/PlayerResultsCard';
+import { PrimaryButton } from '../components/core/Button';
+import { Room } from '../api/Room';
+import ErrorMessage from '../components/core/Error';
+import Loading from '../components/core/Loading';
+import { disconnect, routes, subscribe } from '../api/Socket';
+import { User } from '../api/User';
+
+const Content = styled.div`
+  padding: 0 20%;
+`;
+
+type LocationState = {
+  game: Game,
+  currentUser: User,
+};
 
 function GameResultsPage() {
+  const history = useHistory();
+  const location = useLocation<LocationState>();
+
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const [players, setPlayers] = useState<Player[]>();
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [room, setRoom] = useState<Room | null>(null);
+
+  useEffect(() => {
+    if (checkLocationState(location, 'game', 'currentUser')) {
+      setPlayers(location.state.game.players);
+      setRoom(location.state.game.room);
+      setCurrentUser(location.state.currentUser);
+
+      const subscribeCallback = (result: Message) => {
+        const updatedGame: Game = JSON.parse(result.body);
+
+        // Disconnect users from socket and then redirect them to the lobby page
+        if (updatedGame.playAgain) {
+          disconnect()
+            .then(() => {
+              history.replace(`/game/lobby?room=${updatedGame.room.roomId}`, {
+                user: location.state.currentUser,
+                roomId: updatedGame.room.roomId,
+              });
+            })
```

Comment:
What exactly is happening here -- is a new game being constructed and added to the `currentGameMap`, or does the object just consist of the current game object. It seems like it is the latter, and in that case, how is the game "refreshed" so the information from the previous game does not stay there?

Reply:
It's only passing roomId and the currentUser object to the lobby page (location requirements), so none of the actual game info is carried over (and in fact I think once the other PR is merged in I'll implement the endGame functionality to delete the old game - tho in the future endGame might be to save to the database). And then once on the lobby page it behaves as normal (fetching the RoomDto with the same settings as before, etc.)